pkg/apis/common/v1/association.go

```diff
@@ -48,23 +98,33 @@ type Association interface {
 
 	// AssociatedType returns a string describing the type of the target resource (elasticsearch most of the time)
 	// It is mostly used to build some other strings depending on the type of the targeted resource.
-	AssociatedType() string
+	AssociationType() AssociationType
 
 	// Reference to the associated resource. If defined with a Name then the Namespace is expected to be set in the returned object.
 	AssociationRef() ObjectSelector
 
-	// AssociationConfAnnotationName is the name of the annotation used to define the config for the associated resource.
+	// AssociationConfAnnotationNameBase is the name of the annotation used to define the config for the associated resource.
 	// It is used by the association controller to store the configuration and by the controller which is
 	// managing the associated resource to build the appropriate configuration.
-	AssociationConfAnnotationName() string
+	AnnotationName() string
 
 	// Configuration
 	AssociationConf() *AssociationConf
 	SetAssociationConf(*AssociationConf)
 
-	// Status
-	AssociationStatus() AssociationStatus
-	SetAssociationStatus(status AssociationStatus)
+	// Id allows to distinguish between many associations of the same type
+	ID() int
```

Comment:
I've changed ID() to be a string which actually uniquely identifies Association among all Associations of the same type belonging to Associated().

Given that, I think ID is fine now. Let me know if you think otherwise.

Ordinal is also there, but only for `AgentESAssociation`.